src/components/input/SelectNext.tsx

```diff
@@ -60,7 +60,35 @@ function SelectOption<T>({
   }
 
   const { selectValue, value: currentValue } = selectContext;
-  const selected = !disabled && currentValue === value;
+  const valueIsArray = Array.isArray(currentValue);
```

Comment:
An alternative to this would be testing the `multiple` prop and throwing an error if `currentValue` is not an array in that case, as such a situation is incorrect use of the API.